src/pages/home/report/ReportActionCompose.js

```diff
@@ -166,1137 +179,1146 @@ const getMaxArrowIndex = (numRows, isAutoSuggestionPickerLarge) => {
+                    resetSuggestions();
+                }
 
-        // Trigger the edit box for last sent message if ArrowUp is pressed and the comment is empty and Chronos is not in the participants
-        if (
-            e.key === CONST.KEYBOARD_SHORTCUTS.ARROW_UP.shortcutKey &&
-            this.textInput.selectionStart === 0 &&
-            this.state.value.length === 0 &&
-            !ReportUtils.chatIncludesChronos(this.props.report)
-        ) {
-            e.preventDefault();
-
-            const parentReportActionID = lodashGet(this.props.report, 'parentReportActionID', '');
-            const parentReportAction = lodashGet(this.props.parentReportActions, [parentReportActionID], {});
-            const lastReportAction = _.find([...this.props.reportActions, parentReportAction], (action) => ReportUtils.canEditReportAction(action));
-
-            if (lastReportAction !== -1 && lastReportAction) {
-                Report.saveReportActionDraft(this.props.reportID, lastReportAction.reportActionID, _.last(lastReportAction.message).html);
+                return;
             }
-        }
-    }
 
-    /**
-     * @returns {String}
-     */
-    prepareCommentAndResetComposer() {
-        const trimmedComment = this.comment.trim();
-        const commentLength = ReportUtils.getCommentLength(trimmedComment);
-        // Don't submit empty comments or comments that exceed the character limit
-        if (!commentLength || commentLength > CONST.MAX_COMMENT_LENGTH) {
-            return '';
-        }
+            // Submit the form when Enter is pressed
+            if (e.key === CONST.KEYBOARD_SHORTCUTS.ENTER.shortcutKey && !e.shiftKey) {
+                e.preventDefault();
+                submitForm();
+            }
 
-        this.updateComment('');
-        this.setTextInputShouldClear(true);
-        if (this.props.isComposerFullSize) {
-            Report.setIsComposerFullSize(this.props.reportID, false);
-        }
-        this.setState({isFullComposerAvailable: false});
+            // Trigger the edit box for last sent message if ArrowUp is pressed and the comment is empty and Chronos is not in the participants
+            if (e.key === CONST.KEYBOARD_SHORTCUTS.ARROW_UP.shortcutKey && textInputRef.current.selectionStart === 0 && value.length === 0 && !ReportUtils.chatIncludesChronos(report)) {
+                e.preventDefault();
 
-        return trimmedComment;
-    }
+                const parentReportActionID = lodashGet(report, 'parentReportActionID', '');
+                const parentReportAction = lodashGet(parentReportActions, [parentReportActionID], {});
+                const lastReportAction = _.find([...reportActions, parentReportAction], (action) => ReportUtils.canEditReportAction(action));
+
+                if (lastReportAction !== -1 && lastReportAction) {
+                    Report.saveReportActionDraft(reportID, lastReportAction.reportActionID, _.last(lastReportAction.message).html);
```

Comment:
Sorry I had to hold merge https://github.com/Expensify/App/pull/24544 for this PR.
Now the branch conflicts.
This change should be enough while fixing conflict.
@mountiny can anyone who has write permission to this branch help as @hannojg is OOO?

```suggestion
                    Report.saveReportActionDraft(reportID, lastReportAction, _.last(lastReportAction.message).html);
```

Reply:
asked